float32 dog_length
float32 dog_height
float32 dog_distance
